itenium-socks/src/app/socks/sock.component.ts

```diff
@@ -1,29 +1,69 @@
-import { Component } from '@angular/core';
-import { Observable } from 'rxjs';
-import { Sock } from './sock.model';
-import { SocksService } from './socks.service';
-import { AsyncPipe, NgIf, TitleCasePipe } from '@angular/common';
+import {Component, Input} from '@angular/core';
+import {Observable, catchError, map, tap} from 'rxjs';
+import {Sock} from './sock.model';
+import {SocksService} from './socks.service';
+import {AsyncPipe, NgIf, TitleCasePipe} from '@angular/common';
+import {MatSnackBar} from '@angular/material/snack-bar';
+import {ActivatedRoute} from '@angular/router';
+import {BankAccountModalComponent} from "../components/bank-account-modal/bank-account-modal.component";
+import {
+  MatDialog
+} from '@angular/material/dialog';
 
 @Component({
   selector: 'app-sock',
   standalone: true,
-  imports: [NgIf, AsyncPipe, TitleCasePipe],
-  templateUrl: './sock.component.html'
+  templateUrl: './sock.component.html',
+  imports: [NgIf, AsyncPipe, TitleCasePipe, BankAccountModalComponent]
 })
 export class SockComponent {
+
   sock$!: Observable<Sock>;
+  showAlert: boolean = false;
+  showModal = false;
+  @Input() closedModal = false;
 
-  constructor(private socksService: SocksService) {}
+  constructor(private socksService: SocksService, 
+    private snackBar: MatSnackBar, 
+    private route: ActivatedRoute, 
+    private dialog: MatDialog) { }
 
   ngOnInit(): void {
-    // HACK: This is not the way to get the sockId!!
-    const sockId = +window.location.pathname.split('/')[2];
-    this.sock$ = this.socksService.getById(sockId);
+    this.route.params.pipe(
+      map(params => {
+        const sockId = params['id'];
+        this.sock$ = this.socksService.getById(sockId);
+      })
+    ).subscribe();
```

Comment:
In plaats van 1x een subscribe te doen, probeer een pipeline te declareren die altijd correct is:

```
sock$ = this.route.paramMap.pipe(
    map(params => +(params.get('id') ?? 1)),
    switchMap(sockId => this.socksService.getById(sockId))
)
```